setup.py

```diff
@@ -6,6 +6,13 @@
 from setuptools import setup
 from setuptools import find_packages
 
+
+if not hasattr(sys, 'real_prefix'):
+    exit('Only supported on a Virtual Environment')
+
+if hasattr(os, 'geteuid') and os.geteuid() == 0:
+    exit("Not supported when running as 'root'")
```

Comment:
Similar to above + there is nothing inherently wrong with running this script as root. For instance, on my Let's Encrypt testing VPS, I run virtualenv setup under `root`...